lib/pka_dev.c

```diff
@@ -659,9 +658,19 @@ static int pka_dev_create_shim(pka_dev_shim_t *shim, uint32_t shim_id,
         return -EINVAL;
     }
 
-    shim->shim_id    = shim_id;
-    shim->base       = shim_base;
-    shim->size       = shim_size;
+    shim->shim_id                     = shim_id;

```

Comment:
does it make sense to have a sub-funtion for such initialization?
For example: 
   pka_dev_init_shim_mem_res(pka_dev_shim_t *shim, pka_dev_mem_res* res)

Reply:
We can perhaps use the assign `=` operator to copy two structs here, what do you think ?
If upstream recommends assigning individual members of the struct separately, I shall do so. Let me know.